src/datasets/load.py

```diff
@@ -581,6 +581,7 @@ def __init__(
         self.name = name
         self.revision = revision
         self.download_config = download_config or DownloadConfig()
```

Comment:
you can copy the config here to not modify the user's object if they explicitly passed a DownloadConfig.
```suggestion
        self.download_config = download_config.copy() or DownloadConfig()
```